packages/flutter_tools/test/src/context.dart

```diff
@@ -290,13 +290,11 @@ class FakeDeviceManager implements DeviceManager {
   Device? getSingleEphemeralDevice(List<Device> devices) => null;
 
   List<Device> filteredDevices(DeviceDiscoveryFilter? filter) {
-    if (filter?.deviceConnectionInterface == DeviceConnectionInterface.attached) {
-      return attachedDevices;
-    }
-    if (filter?.deviceConnectionInterface == DeviceConnectionInterface.wireless) {
-      return wirelessDevices;
-    }
-    return attachedDevices + wirelessDevices;
+    return switch (filter?.deviceConnectionInterface) {
+      DeviceConnectionInterface.attached => attachedDevices,
+      DeviceConnectionInterface.wireless => wirelessDevices,
+      null => attachedDevices + wirelessDevices,
```

Comment:
I'm a little curious here as to why `null` is used instead of an arbitrary `_` as in (almost) everywhere else.

Reply:
`Enum`, `bool`, and `sealed` classes are exhaustive—Dart is able to understand when all cases have been covered.

A wildcard pattern `_` in a switch expression (or `default` in a switch statement) goes against the [style guideline](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values) that this pull request was named after.

I made this mistake during my first [switch expressions PR](https://github.com/flutter/flutter/pull/136140#discussion_r1349586844), but never again. 🙂